File: legacy/2026-05-cleanup/wt-editor-routes/engine/pages/regression_page.py
"""RegressionPage — TestwrightAI Regresyon setleri Page Object."""
from playwright.sync_api import Page, expect
from pages.base_page import BasePage


class RegressionPage(BasePage):
    """TestwrightAI Regresyon setleri sayfası."""

    HEADING = '[data-testid="regression-heading"]'
    NAME_INPUT = '[data-testid="regression-input-name"]'
    CREATE_BTN = '[data-testid="regression-btn-create"]'
    SUGGEST_BTN = '[data-testid="regression-btn-suggest"]'

    def __init__(self, page: Page, project_id: str, base_url: str = ""):
        super().__init__(page)
        self._project_id = project_id
        self._base_url = base_url

    def goto(self):
        self.navigate(f"{self._base_url}/p/{self._project_id}/regression")
        return self

    def create_set(self, name: str):
        self.page.locator(self.NAME_INPUT).fill(name)
        self.page.locator(self.CREATE_BTN).click()
        return self

    def suggest(self):
        self.page.locator(self.SUGGEST_BTN).click()
        return self

    def assert_page_loaded(self):
        expect(self.page.locator(self.HEADING)).to_be_visible()
        return self

    def assert_set_visible(self, name: str):
        expect(self.page.get_by_text(name)).to_be_visible(timeout=10_000)
        return self

File: e2e/pages/regression.page.ts
import { expect } from "@playwright/test";
import { BasePage } from "./base.page";

export class RegressionPage extends BasePage {
  readonly url: RegExp = /\/p\/[^/]+\/regression$/;

  // ── Locators ───────────────────────────────────────────────────────────────
  get newSetButton() {
    return this.page.getByTestId("regression-page").getByRole("button", { name: "Yeni Set" });
  }
  get nameInput() {
    return this.page.locator("#regression-set-name");
  }
  get saveButton() {
    return this.page.getByTestId("regression-page").getByRole("button", { name: "Kaydet" });
  }
  get suggestButton() {
    return this.role("button", { name: /öner|suggest/i });
  }
  get checkboxes() {
    return this.role("checkbox");
  }

  // ── Actions ────────────────────────────────────────────────────────────────
  async createSet(name: string, scenarioCount = 2) {
    await expect(this.page.getByTestId("regression-page")).toBeVisible({ timeout: 10_000 });
    await this.newSetButton.click();
    await this.nameInput.fill(name);
    const count = await this.checkboxes.count();
    for (let i = 0; i < Math.min(count, scenarioCount); i++) {
      await this.checkboxes.nth(i).check();
    }
    await this.saveButton.click();
  }

  async aiSuggest() {
    await expect(this.suggestButton).toBeVisible({ timeout: 10_000 });
    await this.suggestButton.click();
  }

  // ── Assertions ─────────────────────────────────────────────────────────────
  async assertSetVisible(name: string) {
    await expect(this.text(name)).toBeVisible({ timeout: 10_000 });
  }

  async assertSuggestionVisible() {
    await expect(
      this.text(/öneri|suggestion|senaryo/i).first()
    ).toBeVisible({ timeout: 30_000 });
  }
}
